src/effects/VST/VSTEffect.cpp

```diff
@@ -1038,7 +1037,14 @@ bool VSTEffect::ProcessFinalize()
    return true;
 }
 
-size_t VSTEffect::ProcessBlock(EffectSettings &,
+
+bool VSTEffectInstance::ProcessFinalize()
+{
+   return VSTEffectWrapper::DoProcessFinalize();
```

Comment:
I don't like `DoProcessFinalize`, `DoProcessInitialize`, or `ProcessBlockB`.  Too much is stil being done in the wrapper class, and therefore the Effect still has more state that it should.  The purpose of this PR should be to remove as much effect state as possible, identifying the parts that should be per-instance state.

And I think the reason you thought you needed this is:  you let the slaves remain VSTEffect objects, and did not change the type of the slaves to be VSTEffectInstance.  But you can do this.

The array of slaves remains temporarily in the Effect, until step 5 is done.  But the virtual functions needed in the slaves are only `ProcessFinalize` and `ProcessBlock`, now in the Instance class, and other functions already in the Wrapper.

Reply:
I now implemented the "middle class" VstInstanceBase as we discussed, please have a look there.

Reply:
I do not yet see a force-push with that update